bolero/controller/controller.py

```diff
@@ -308,6 +309,120 @@ def _perform_test(self, meta_parameter_keys, meta_parameters):
         return performance - optimum
 
 
+class AveragingEpisodesController(Controller):
+    """Controller for non-deterministic problems.
+
+    See base class "Controller" for details on usage.
+
+    Additional Parameters
+    ----------
+    num_repetitions_to_average : int, optional (default: 10)
+        Number of repetitions per episode. For one episode, several rollouts
+        are executed with the same behavior. Note that this only makes sense
+        if the environment is stochastic or specifically prepared via the
+        argument environment_preparation_function
+
+    feedback_averaging_function : function, optional (default: median_of_sums)
```

Comment:
It is a callback, not just a function. It also does not have to be a function, it can be any callable.